lib/name.ml

```diff
@@ -93,15 +93,16 @@ let parse names base buf = (* what. a. mess. *)
           aux (o :: offsets) (n :: name) (base+offset) (Cstruct.shift buf offset)
 
       | (P (p, _), offset) ->
-          let ns = (Hashtbl.find_all names p
-                       |> List.filter (function L _ -> true | _ -> false))
-          in
+          let all = Hashtbl.find_all names p in
+          if ((List.length all) == 0) then
```

Comment:
you don't need the parenthesis around the condition, and `List.length` is an expensive operation.

What about:
```OCaml
(match all with
 | [] -> failwith (sprintf "Name.parse_pointer...")
 | _ -> ...rest of body...)
```